src/nvim/eval.c

```diff
@@ -10982,81 +10982,123 @@ static int inputsecret_flag = 0;
  */
 static void get_user_input(typval_T *argvars, typval_T *rettv, int inputdialog)
 {
-  const char *prompt = tv_get_string_chk(&argvars[0]);
-  int cmd_silent_save = cmd_silent;
-  int xp_type = EXPAND_NOTHING;
-  char_u      *xp_arg = NULL;
-
   rettv->v_type = VAR_STRING;
   rettv->vval.v_string = NULL;
 
-  cmd_silent = FALSE;           /* Want to see the prompt. */
-  if (prompt != NULL) {
-    // Only the part of the message after the last NL is considered as
-    // prompt for the command line.
-    const char *p = strrchr(prompt, '\n');
-    if (p == NULL) {
-      p = prompt;
-    } else {
-      p++;
-      msg_start();
-      msg_clr_eos();
-      msg_puts_attr_len(prompt, p - prompt, echo_attr);
-      msg_didout = false;
-      msg_starthere();
+  const char *prompt = "";
+  const char *defstr = "";
+  const char *cancelreturn = NULL;
+  const char *xp_name = NULL;
+  char prompt_buf[NUMBUFLEN];
+  char defstr_buf[NUMBUFLEN];
+  char cancelreturn_buf[NUMBUFLEN];
+  char xp_name_buf[NUMBUFLEN];
+  if (argvars[0].v_type == VAR_DICT) {
+    if (argvars[1].v_type != VAR_UNKNOWN) {
+      emsgf(
+          _("E5050: When providing {opts} argument no more arguments follow"));
```

Comment:
This makes it sound like extra args are required. Suggestion:

    E5050: {opts} must be the only argument